components/renku_data_services/session/k8s_client.py

```diff
@@ -198,3 +137,54 @@ async def get_server(self, name: str) -> BuildRun | None:
                 message=f"Expected to find 1 build run when getting run {name}, found {len(output)}."
             )
         return BuildRun.model_validate(output[0])
+
+
+class ShipwrightClient:
+    """The K8s client that combines a base client and a cache."""
+
+    def __init__(
+        self,
+        cache: ShipwrightCache | None,
+        base_client: ShipwrightClientBase,
+        # NOTE: If cache skipping is enabled then when the cache fails a large number of
+        # buildruns can overload the k8s API by submitting a lot of calls directly.
+        skip_cache_if_unavailable: bool = False,
+    ) -> None:
+        self.cache = cache
+        self.base_client = base_client
+        self.skip_cache_if_unavailable = skip_cache_if_unavailable
+
+    async def list_build_runs(self) -> list[BuildRun]:
+        """Get a list of ShipWright BuildRuns."""
+        if self.cache is None:
+            return await self.base_client.list_build_runs()
```

Comment:
issue: Fallback to cache only when the `skip_cache...` flag is set to `True`.

Reply:
The cache is not deployed in this PR and will not be considered functional in this PR. As such, the cache will always be skipped for now.